src/content/docs/en/guides/server-islands.mdx

```diff
@@ -71,6 +71,14 @@ Each island is loaded independently from the rest. This means a slower island wo
 
 This rendering pattern was built to be portable. It does not depend on any server infrastructure so it will work with any host you have, from a Node.js server in a Docker container to the serverless provider of your choice.
 
+## Caching
+
+Generally, it is possible to cache server islands since their data is retrieved via a `GET` request. This can be done with the [`Cache-Control` HTTP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) and follows the default behavior. 
+
+However, since Astro passes the props as an encrypted string in the query, in cases where the URL becomes larger than 2048 bytes [the request will be ignored by the browser](https://chromium.googlesource.com/chromium/src/+/master/docs/security/url_display_guidelines/url_display_guidelines.md#url-length). In such cases, Astro will send a `POST`  request instead that contains all props in the body. 
+
+`POST` requests are not cached by all major browsers due to their common use for forms, so this will break any caching logic you have set up. To avoid this issue, try not to pass any big props like objects or arrays to your server islands.
```

Comment:
```suggestion
The data for server islands is retrieved via a `GET` request, passing props as an encrypted string in the URL query. This allows caching data with the [`Cache-Control` HTTP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) using standard `Cache-Control` directives.

However, [the browser limits URLs to a maximum length of 2048 bytes](https://chromium.googlesource.com/chromium/src/+/master/docs/security/url_display_guidelines/url_display_guidelines.md#url-length) for practical reasons and to avoid causing denial-of-service problems. If your query string causes your URL to exceed this limit, Astro will instead send a `POST` request that contains all props in the body. 

`POST` requests are not cached by browsers because they are used to submit data, and could cause data integrity or security issues. Therefore, any existing caching logic in your project will break. Whenever possible, pass only necessary props to your server islands and avoid sending entire data objects and arrays to keep your query small.
```
Took a shot at this. What do you think?

Reply:
Sounds good!